### PR TITLE
Display progress information

### LINKED ISSUE #594: Display Borg Progress data, different progress indicator
**Is your feature request related to a problem? Please describe.**
It's very annoying when one runs a bigger backup, which will take some time, and there's absolutely no visible progress and ETA.

**Describe the solution you'd like**
- Change the progress bar on the "Repository" tab to actually show the progress in percent. For an easy solution, simply count the files, ignoring their size, that's better than it is now.
- Add status text, containing the following: Current speed, bytes copied, bytes left, and depending on the speed, give an ETA (Yes, this might be inaccurate, I'm aware of that)

Disclaimer:
I'm not familiar with the borg API, so probably there's no (easy) way to get progress/files count/data sizes etc. If so, maybe some changes could be made to borg first?

**Describe alternatives you've considered**
Can't find one

### PR DESCRIPTION
**Pull request by SergioRAgostinho**
An implementation of what was described in 
https://github.com/borgbase/vorta/issues/594#issuecomment-686363495

See it live
output

Closes #594 

On an unrelated note: the repo stats reporting a deduplicated size larger than the compressed size contradicts a bit the information from the progress report. Is this expected?

### CODE REVIEW

File: src/vorta/borg/borg_thread.py [hunk 1]
```diff
@@ -200,6 +200,14 @@ def read_async(fd):
                             logger.log(level_int, parsed["message"])
                         elif parsed['type'] == 'file_status':
                             self.log_event(f'{parsed["path"]} ({parsed["status"]})')
+                        elif parsed['type'] == 'archive_progress':
+                            msg = (
+                                f'Original: {pretty_bytes(parsed["original_size"])}, '
+                                f'Compressed: {pretty_bytes(parsed["compressed_size"])}, '
+                                f'Deduplicated: {pretty_bytes(parsed["deduplicated_size"])}, '
```
Comment (SergioRAgostinho) [inline src/vorta/borg/borg_thread.py:207]: Potentially change order here to match the repo-tab order.

**Comment by XXXBold** [comment]
One more thing: It would be nice if the total file count could also be shown, if available, means extend the string at the end to "...Files: 10/1234"

**Comment by SergioRAgostinho** [comment]
That information is not available. Here's a sample of the json payload.
```
{'original_size': 229620919, 'compressed_size': 37674787, 'deduplicated_size': 37674787, 'nfiles': 205, 'time': 1599160038.8519056,  'type': 'archive_progress', 'path': 'tmp/data/ModelNet40/curtain/train/curtain_0130.off'} 
```

**Comment by XXXBold** [comment]
Okay, too bad, but I might have another method: Will it help if you can get the current files count from the sources control? Or do you need another number?

I'm asking cause I'm working on getting the directory/file sizes within the source list, and if it helps you I could also simply count all files/folders.

**Comment by SergioRAgostinho** [comment]
> I'm asking cause I'm working on getting the directory/file sizes within the source list, and if it helps you I could also simply count all files/folders.

Is there something already implemented?

If yes, can you point me to the method? I can cache this at the beginning of the backup and use it not only to report the number but to update the progress bar value as well.

If not, can we move this to a different PR?

**Comment by XXXBold** [comment]
It's not yet available, but I'm working on it, see #593 point no. 4, will take a while, though.
I guess using for the progress bar could also be added later, so you don't have to wait with this PR.

**Comment by m3nu** [comment]
Wow. Love it. 😍

**Comment by m3nu** [comment]
MacOS can't have text inside the progress bar. Maybe we use a different element instead of it.

> Note that whether or not the text is drawn is dependent on the style. Currently CleanLooks and Plastique draw the text. Mac, Windows and WindowsXP style do not. https://doc.qt.io/archives/qt-5.8/qprogressbar.html#text-prop

**Comment by m3nu** [comment]
Screen Shot 2020-09-04 at 14 52 09

Using a label field, so it works everywhere.

**Comment by m3nu** [comment]
Basic GIF animation to replace progress bar.

Screen Shot 2020-09-04 at 15 26 20

**Comment by m3nu** [comment]
We should probably call this primary- and secondary status display and use it for different things. E.g. the primary one could display our own status messages, while secondary shows Borg's logs.

**Comment by SergioRAgostinho** [comment]
> Basic GIF animation to replace progress bar.

It's looking a bit static :sweat_smile:

**Comment by m3nu** [comment]
Because I'm too lazy to make a movie, like you. But it does do something.

**Comment by SergioRAgostinho** [comment]
How it currently looks on my setup
output


> Because I'm too lazy to make a movie, like you.

Don't let that stop you. Here's the one-liner in case you already have ffmpeg installed
```
ffmpeg -i input-video.mp4 -vf "fps=15,scale=640:-1:flags=lanczos,split[s0][s1];[s0]palettegen[p];[s1][p]paletteuse" -loop 0 output.gif
```

**Comment by SergioRAgostinho** [comment]
:+1: I will not touch this one then. I think it's pretty much finished if we exclude the potential ui revision you might want to perform now, with the two existing text labels.

**Comment by m3nu** [comment]
Closing in favor of #615

**Pull request closed without merging**